src/qtfirebaseadmob.cpp

```diff
@@ -1123,8 +1132,10 @@ firebase::FutureBase QtFirebaseAdMobRewardedVideoAd::initInternal()
     return firebase::FutureBase();
 }
 
-void QtFirebaseAdMobRewardedVideoAd::onPresentationStateChanged(int state)
+void QtFirebaseAdMobRewardedVideoAd::setPresentationState(QtFirebaseAdMobInterstitial::PresentationState state)
```

Comment:
`RewardedVideoAd` shouldn't use `AdMobInterstitial` ENUMS. I'm sure a re-factoring and splitting classes into their own files help with the overview.

I'd like we keep separate ENUMS for each ad type - in case Google decides to expand with something new and fancy in one place + it keeps things separated which is better for readability. I'm trying to find time for splitting the everything into it's own files for better structure and readability - then we can look at a "rewrite" from there as you issued.